Fix prob cache bug

CGLemon opened the pull request:
The value may be mistake from cache because the NNCache Entry did not record the komi. If we set the different komi with cache and got the value from prob cache at the same time, it would get the wrong value.

CGLemon committed:
Merge pull request #1 from sai-dev/sai-0.17

Merge

CGLemon committed:
Fix prob cache bug

Vandertic reviewed (changes requested):
I think now the `if` of line 987 is not necessary anymore. Can you change lines 986--989 to something like
`bool cache_success = m_nncache.lookup(...);`

Vandertic commented:
BTW thanks for the help! 😲

CGLemon committed:
Fix probe cache pipeline

Simplify and fix the bug that it does not out of loop even if the cache is success.

CGLemon commented:
Done
